include/directx/d3d12shader.h

```diff
@@ -169,7 +169,7 @@ typedef struct _D3D12_SHADER_INPUT_BIND_DESC
     UINT                        uFlags;         // Input binding flags
     D3D_RESOURCE_RETURN_TYPE    ReturnType;     // Return type (if texture)
     D3D_SRV_DIMENSION           Dimension;      // Dimension (if texture)
-    UINT                        NumSamples;     // Number of samples (0 if not MS texture)
+    UINT                        NumSamples;     // Number of samples (0 if MS texture, UINT_MAX otherwise)

```

Comment:
This isn't quite right. This value can be:
* 0 if the texture is a multisampled texture that doesn't explicitly declare a sample count, or for raw buffers.
* A value <= 32 if a sample count is declared in the shader.
* An arbitrary value != `UINT_MAX` for structured buffers, to report structure byte stride.
* `UINT_MAX` for a texture that's not multisampled.

Reply:
As a note:
- RTAS also has UINT_MAX.
- CBuffer/BAB are included in "buffers" aka 0.
- SamplerState is also 0 (unless there's a feature I missed).
- As said before; sbuffer includes stride there.